### PR TITLE
The player the floats through space!

### PR DESCRIPTION
**Pull request by Ashthecool**
The player floats through space, it glides and wont stop until you turn or press down, with is a break. The player will go the direction you are facing. Got the information from here: [https://api.arcade.academy/en/latest/examples/sprite_move_angle.html#sprite-move-angle](url)

### CODE REVIEW

**Review by tobiasbp — CHANGES_REQUESTED** [review]
We don't want your .idea dir in our code. It's a folder used by your local pycharm.
Changes in the PR should be for my_game.py only.

**Comment by Ashthecool** [comment]
> We don't want your .idea dir in our code. It's a folder used by your local pycharm. Changes in the PR should be for my_game.py only.

How do I do that? T.T

**Comment by Ashthecool** [comment]
>

File: my_game.py [hunk 1]
```diff
@@ -44,20 +44,24 @@ def __init__(self, **kwargs):
         # Pass arguments to class arcade.Sprite
         super().__init__(**kwargs)
 
+        self.speed = 0
+        self.change_x = -math.sin(math.radians(self.angle)) * self.speed
+        self.change_y = math.cos(math.radians(self.angle)) * self.speed
+        self.max_speed = 5
+        self.thrust = 0
+        self.drag = 0.05
```
Comment (tobiasbp) [inline my_game.py:52]: No drag in space.

File: my_game.py [hunk 2]
```diff
@@ -110,6 +114,9 @@ def __init__(self, width, height):
         self.player_sprite = None
         self.player_score = None
         self.player_lives = None
+        self.player_speed = 0
+        self.drag = 0.05
+        self.opposite_angle = 0
```
Comment (tobiasbp) [inline my_game.py:119]: Sounds like stuff which should be in the Player class.

File: my_game.py [hunk 3]
```diff
@@ -158,8 +164,8 @@ def setup(self):
 
         # Create a Player object
         self.player_sprite = Player(
-            center_x=PLAYER_START_X,
-            center_y=PLAYER_START_Y
+            center_x=SCREEN_HEIGHT//2,
+            center_y=SCREEN_WIDTH//2
```
Comment (tobiasbp) [inline my_game.py:168]: Don't change these. It's not part of your feature.

File: my_game.py [hunk 4]
```diff
@@ -18,13 +18,13 @@
 
 # Variables controlling the player
 PLAYER_LIVES = 3
-PLAYER_SPEED_X = 5
-PLAYER_START_X = SCREEN_WIDTH / 2
-PLAYER_START_Y = 50
```
Comment (tobiasbp) [inline my_game.py:23]: Don't remove these. It's not part of your feature.

File: my_game.py [hunk 5]
```diff
@@ -178,36 +184,47 @@ def on_draw(self):
 
         # Draw players score on screen
         arcade.draw_text(
-            "SCORE: {}".format(self.player_score),  # Text to show
-            10,                  # X position
-            SCREEN_HEIGHT - 20,  # Y positon
-            arcade.color.WHITE   # Color of text
+            "SCORE: {}".format(self.opposite_angle),  # Text to show
```
Comment (tobiasbp) [inline my_game.py:187]: Angle should not be showed as score. Please remove the above changes.

File: my_game.py [hunk 6]
```diff
@@ -178,36 +184,47 @@ def on_draw(self):
...
-        # Move player with keyboard
-        if self.left_pressed and not self.right_pressed:
-            self.player_sprite.change_x = -PLAYER_SPEED_X
-        elif self.right_pressed and not self.left_pressed:
-            self.player_sprite.change_x = PLAYER_SPEED_X
-
         # Move player with joystick if present
         if self.joystick:
-            self.player_sprite.change_x = round(self.joystick.x) * PLAYER_SPEED_X
+            self.player_sprite.change_x = round(self.joystick.x) * 4
```
Comment (tobiasbp) [inline my_game.py:200]: Don't do this. We want to keep using that variable.

File: my_game.py [hunk 7]
```diff
@@ -178,36 +184,47 @@ def on_draw(self):
...
+
+        if not self.up_pressed:
+            if not self.player_sprite.speed < 0.05:
+                if self.down_pressed or self.right_pressed or self.left_pressed:
+                    self.player_sprite.speed -= 0.1
+                else:
+                    self.player_sprite.speed -= 0.001
+        else:
+            if not self.player_sprite.speed > 6:
+                self.player_sprite.speed += 0.05
```
Comment (tobiasbp) [inline my_game.py:226]: We don't want magic numbers here. Please use variables. I'm pretty sure you added those values to the Player class in lines 50 to 52.

**Review by tobiasbp — CHANGES_REQUESTED** [review]
Take a look at my comments.

File: my_game.py [hunk 8]
```diff
@@ -44,20 +44,24 @@ def __init__(self, **kwargs):
         # Pass arguments to class arcade.Sprite
         super().__init__(**kwargs)
 
+        self.speed = 0
+        self.change_x = -math.sin(math.radians(self.angle)) * self.speed
+        self.change_y = math.cos(math.radians(self.angle)) * self.speed
+        self.max_speed = 5
```
Comment (tobiasbp) [inline my_game.py:50]: max_speed is never used?

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.py [hunk 9]
```diff
@@ -208,6 +203,10 @@ def on_update(self, delta_time):
         # Update the player shots
         self.player_shot_list.update()
 
+        if self.up_pressed:
+            if not self.player_sprite.speed > self.max_speed:
+                self.player_sprite.speed += 0.05
```
Comment (tobiasbp) [inline my_game.py:208]: The 0.05 need to go in to a variable. Let's make a global variable PLAYER_THRUST, and use that here.

At beginning of file:
PLAYER_THRUST = 0.05

Then use that variable here.

File: my_game.py [hunk 10]
```diff
@@ -110,8 +111,12 @@ def __init__(self, width, height):
         self.player_sprite = None
         self.player_score = None
         self.player_lives = None
+        self.player_speed = 0
+        self.opposite_angle = 0
+        self.max_speed = 3
 
         # Track the current state of what key is pressed
+        self.space_pressed = None
```
Comment (tobiasbp) [inline my_game.py:119]: Should be _False_, non _None_.

File: my_game.py [hunk 11]
```diff
@@ -17,14 +17,17 @@
 SCREEN_HEIGHT = 600
 
 # Variables controlling the player
-PLAYER_LIVES = 3
 PLAYER_SPEED_X = 5
 PLAYER_START_X = SCREEN_WIDTH / 2
 PLAYER_START_Y = 50
+PLAYER_LIVES = 3
+PLAYER_SPEED = 3
```
Comment (tobiasbp) [inline my_game.py:24]: _PLAYER_SPEED_ is not used. Please remove it. I think your editor also warns you about this.

**Review by tobiasbp — CHANGES_REQUESTED** [review]

**Review by tobiasbp — APPROVED** [review]
Looks good. Now, let's merge it it.

**Pull request merged**